src/Sylius/Component/Promotion/Generator/CouponGenerator.php

```diff
@@ -91,6 +113,33 @@ public function generateUniqueCode()
      */
     protected function isUsedCode($code)
     {
-        return null !== $this->repository->findOneBy(['code' => $code]);
+        foreach ($this->generatedCoupons as $coupon) {
```

Comment:
This is very bad performance wise. If you have 10,000 coupons of length 10 and you want to generate 10,000 more - you will hydrate 10,000 and flush 20,000 entities. Better solution would be to return array of coupon codes (not objects, strings) from repository and simply search in array.

This foreach will iterate over all coupons * amount of coupons you want to generate. Crazy.